Helper/Marketplace/WebkulHelper.php

```diff
@@ -59,55 +61,81 @@ public function getSplitDataFromOrder($platformOrder)
     {
         $orderItems = $platformOrder->getAllItems();
         $splitData['sellers'] = [];
-        $totalPaid = (int) $this->moneyService->floatToCents(
-            $platformOrder->getGrandTotal()
-        );
+        $splitData['marketplace']['totalCommission'] = 0;
+        $totalPaid = $platformOrder->getSubTotal();
+        $totalPaidProductWithoutSeller = 0;
 
         foreach ($orderItems as $item) {
             $productId = $item->getProductId();
             $sellerDetail = $this->mpPaymentHelper->getSellerDetail($productId);
             $sellerId = $sellerDetail['id'];
             $itemPrice = $item->getRowTotal();
 
-            if(!$sellerId) {
+            if (!$sellerId) {
+                $totalPaidProductWithoutSeller += $itemPrice;
                 continue;
             }
 
-            $sellerDataForProduct = $this->getSellerData(
+            $dataForProduct = $this->getProductData(
                 $itemPrice,
                 $sellerDetail
             );
 
             if (array_key_exists($sellerId, $splitData['sellers'])) {
-                $splitData['sellers'][$sellerId]['sellerCommission']
-                    += $sellerDataForProduct['sellerCommission'];
+                $splitData['sellers'][$sellerId]['commission']
+                    += $dataForProduct['commission'];
                 continue;
             }
 
-            $splitData['sellers'][$sellerId] = $sellerDataForProduct;
+            $splitData['sellers'][$sellerId] = $dataForProduct;
+            $splitData['marketplace']['totalCommission']
+                += $dataForProduct['marketplaceCommission'];
         }
 
-        if(empty($splitData['sellers'])) {
+        if (empty($splitData['sellers'])) {
             return null;
         }
 
-        $marketplaceCommission
-            = $totalPaid - $this->getTotalSellerCommission(
-                $splitData['sellers']
-            );
-        $splitData['marketplace']
-            = ['marketplaceCommission' => $marketplaceCommission];
+        $totalSellerCommission = $this->moneyService->floatToCents(
```

Comment:
Não faz mais sentido usar o método `floatToCentsSplitData` uma só vez pra formatar todos os valores dentro da variável `$splitData` e daí só usar os valores corretamente formatados? Assim a gente evitaria algumas linhas onde a gente refaz essa conversão usando o método `$this->moneyService->floatToCents`

Reply:
corrigido!